app/controllers/internal/events_controller.rb

```diff
@@ -0,0 +1,140 @@
+module Internal
+  class EventsController < ::InternalController
+    before_action :load_buildings, only: %i[new edit create update]
+    layout "internal"
+
+    def index
+      @no_results = load_pending_events.blank?
+    end
+
+    def show
+      @event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @page_title = @event.name
+      if @event.status_id != GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+        raise ActionController::RoutingError, "Not Found"
```

Comment:
There's a helper for this you should just be able to call:

```
render_not_found unless @event.status_id == pending_event_status_id
```

^ worth pulling out the constant into a private helper method IMO as its reused and quite a mouthful